### ISSUE TITLE
Simple Linear Regression under PyTorch diverges

### ISSUE THREAD

**Issue by SG-391** [pytorch/pytorch#23142]
Hi,
I am using a very basic linear regression, using code suggested on a course as per below, but get very different result. Any idea why this is diverging? 

Thanks for your help!
SG

```
X = torch.randn(100, 1)*10
y = X + 3*torch.randn(100, 1)

class LR(nn.Module):
  def __init__(self, input_size, output_size):
    super().__init__()
    self.linear = nn.Linear(input_size, output_size)
  def forward(self, x):
    pred = self.linear(x)
    return pred

torch.manual_seed(1)
model = LR(1, 1)

criterion = nn.MSELoss()
optimizer = torch.optim.SGD(model.parameters(), lr = 0.01)
print(optimizer.state_dict())

The print gives me => {'state': {}, 'param_groups': [{'lr': 0.01, 'momentum': 0, 'dampening': 0, 'weight_decay': 0, 'nesterov': False, 'params': [1865908224792, 1865911284072]}]}

epochs = 10
for i in range(epochs):
  y_pred = model.forward(X)
  loss = criterion(y_pred, y)
  print("epoch:", i, "loss:", loss.item())
  optimizer.zero_grad()
  loss.backward()
  optimizer.step()
```

I get the following results:
epoch: 0 loss: 36.44832992553711
epoch: 1 loss: 43.94552230834961
epoch: 2 loss: 53.461822509765625
epoch: 3 loss: 65.54042053222656
epoch: 4 loss: 80.87049865722656
epoch: 5 loss: 100.32696533203125
epoch: 6 loss: 125.01981353759766
epoch: 7 loss: 156.35780334472656
epoch: 8 loss: 196.1287384033203
epoch: 9 loss: 246.6014404296875

**Comment by SsnL** [comment]
Can you provide the env information **as suggested by the issue template**?

This is what I got from my machine:
```
epoch: 0 loss: 24.99378204345703
epoch: 1 loss: 19.329004287719727
epoch: 2 loss: 15.663400650024414
epoch: 3 loss: 13.291433334350586
epoch: 4 loss: 11.756567001342773
epoch: 5 loss: 10.763375282287598
epoch: 6 loss: 10.120688438415527
epoch: 7 loss: 9.70481014251709
epoch: 8 loss: 9.435697555541992
epoch: 9 loss: 9.261553764343262
```

So I suspect that something is wrong with MKL/MKLDNN.

**Comment by XiaobingSuper** [comment]
@SsnL , I find that the divergence of loss is happend randomly, you can run the test script many times, and then you can see the problem.

**Comment by mfuntowicz** [comment]
On master b8c8977be797cb16354aa97c3fcdc910d3e9e5c5 it seems to diverge too

epoch: 0 loss: 35.40628433227539
epoch: 1 loss: 44.1915283203125
epoch: 2 loss: 56.21075439453125
epoch: 3 loss: 72.6522445678711
epoch: 4 loss: 95.1410903930664
epoch: 5 loss: 125.89936065673828
epoch: 6 loss: 167.96603393554688
epoch: 7 loss: 225.4969482421875
epoch: 8 loss: 304.1747131347656
epoch: 9 loss: 411.77142333984375
epoch: 10 loss: 558.914306640625
epoch: 11 loss: 760.1366577148438
epoch: 12 loss: 1035.313232421875
epoch: 13 loss: 1411.6219482421875
epoch: 14 loss: 1926.2279052734375
epoch: 15 loss: 2629.958251953125
epoch: 16 loss: 3592.318359375
epoch: 17 loss: 4908.35498046875
epoch: 18 loss: 6708.04345703125
epoch: 19 loss: 9169.1328125
epoch: 20 loss: 12534.6845703125

**Comment by SG-391** [comment]
Environment is the following - Windows, CPU Xeon Silver 4114, Python 3.7.3, PyTorch version 1.1.0, Jupiter notebook

**Comment by SsnL** [comment]
If it is random, it could be that the LR is too big.

**Comment by SG-391** [comment]
My result was actually not random, very consistent instead. But it does work with lr = 0.01. Thanks! 

Any idea as to why two environments may have such different results for the same lr / same model?

**Comment by illcat** [comment]
You should writing "torch.manual_seed(1)"  in front.
`
import torch
import torch.nn as nn
torch.manual_seed(1)
`

**Comment by soumith** [comment]
I dont see the problem if this is caused by a high learning rate.

As others have pointed out, you don't fix the seed before initializing the weights, so every run is seeing different weights.

Have you tried reducing your learning rate?

**Comment by SG-391** [comment]
Yes it does work with a smaller LR. Thanks!

**Comment by Theocrat** [comment]
So for learning rates smaller than 0.01, the loss is guaranteed to not diverge?
